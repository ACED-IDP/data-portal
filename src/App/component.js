import React from 'react';
import { Box } from '../theme';
import Nav from '../Nav/component.js'
import { connect } from 'react-redux';
import styled from 'styled-components';
import { Link } from 'react-router';
import { BoxWithNavAndTimeout } from '../component';

const ProjectLink = styled(Link)`
  cursor: pointer;
  li;
  background: ghostwhite;
  padding: 5px;
  border-radius: 5px;
  display: inline-block;
  margin-bottom: 1em;
  margin-right: 1em;
  color: #8a6d3b;
  &:hover,
  &:focus,
  &:active {
    border: 1px solid #8a6d3b;
    color: #8a6d3b;
  }
`;

const SubmissionComponent = ({submission}) => {
  return (
    <BoxWithNavAndTimeout>
      <h3>Submission projects</h3>
      <ul>
        {submission.projects &&
          <div>
           {submission.projects.map((project) => {return <ProjectLink to={'/'+project} key={project}>{project}</ProjectLink>})}
          </div>
        }
      </ul>
    </BoxWithNavAndTimeout>
  )

};

const mapStateToProps = (state) => {
  return {
    'submission': state.submission
  }
};

const mapDispatchToProps = (dispatch) => {
  return {};
};


let Submission = connect(mapStateToProps, mapDispatchToProps)(SubmissionComponent);
export default Submission;
