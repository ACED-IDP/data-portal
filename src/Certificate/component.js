import React, {Component} from 'react';
import { Form, FormError, RadioGroup, Radio } from 'react-form';
import Nav from '../Nav/component.js'
import { submitForm, updateForm } from './actions';
import { connect } from 'react-redux';
import { Box } from '../theme';
import { QuestionItem, OptionBullet, SubmitButton, Tooltip } from './style';
import * as constants from "./constants";

class Question extends Component {
  render() {
    return (
      <QuestionItem>
        <div>
          <span>{this.props.content.question}</span>
          <Tooltip className='fui-question-circle'>
            <p>{this.props.content.hint}</p>
          </Tooltip>
          { this.props.showErrors && <FormError field={this.props.content.id} /> }
        </div>
        <RadioGroup showErrors={false} field={this.props.content.id}>
        {this.props.content.options.map((option, i) =>
          {
            return <OptionBullet key={i}><Radio onChange={this.props.onChange} name={this.props.content.id} value={option}/>{option}</OptionBullet>
          }
        )}
        </RadioGroup>
      </QuestionItem>
    )
  }

}

class CertificateComponent extends Component {
  constructor() {
    super();
    this.state = {display_error: false};
    // required to be able to pass to child
    this.hideError = this.hideError.bind(this);
  }
  validateForm  (values) {
    // update redux store
    if (Object.keys(values).length > 0){
      this.props.onUpdateForm(values);
    }
    let result = {};
    for (var item of constants.certificate_form){
      result[[item.id]] = values[item.id] === item.options[item.answer] ? undefined: 'wrong answer';
    }
    return result;
  }
  // hide errors when user is updating the form
  hideError() {
    this.setState({display_error: false});
  }

  // show errors when user hits submit button
  showError() {
    this.setState({display_error: true});
  }
  render() {
    return (
      <Box>
        <Nav />
        <h4>{constants.title}</h4>
       <Form onSubmit={(values)=> {this.props.onSubmitForm(values)}}
          validate={(values) => {return this.validateForm(values)}}
          defaultValues={this.props.certificate.certificate_result}>

          {({errors, submitForm}) => {
            return (
              <form>
                {
                  constants.certificate_form.map((item, i) => <Question content={item} onChange={this.hideError} index={i} key={i} showErrors={this.state.display_error}/>)
                }
                <SubmitButton type='submit' onClick={()=> {submitForm(); this.showError();}}>Submit</SubmitButton>
              </form>
            )
          }}

        </Form>
      </Box>
    )
    }
}

const mapStateToProps = (state) => {
  return {
    user: state.user,
    certificate: state.certificate
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    onUpdateForm: (data) => dispatch(updateForm(data)),
    onSubmitForm: (data) => dispatch(submitForm(data)),
  }
}

let Certificate = connect(mapStateToProps, mapDispatchToProps)(CertificateComponent);
export default Certificate;
