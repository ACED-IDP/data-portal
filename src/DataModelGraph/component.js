import React from 'react';
import CreateGraph from './GraphCreator';
import { connect } from 'react-redux';
import { getCounts } from './actions';
import { submissionapi_path } from '../localconf';
import { button } from '../theme';
import styled from 'styled-components';
import { createNodesAndEdges } from '../utils'

const ToggleButton = styled.a`
  border: 1px solid darkslategray;
  color: darkslategray;
  ${button};
  position:absolute;
  top:15px;
  left:20px;
  &:hover,
  &:active,
  &:focus {
    color: black;
    border-color: black;
  }
`;


class DataModelGraphComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      compact: undefined,
      full: undefined, 
      full_toggle: false
    }
    this.state.compact = createNodesAndEdges(props, this.state.full_toggle);
    this.handleClick = this.handleClick.bind(this);
  }
  componentWillReceiveProps(nextProps) {
    if (this.state.full_toggle) {
      this.state.full = createNodesAndEdges(nextProps, this.state.full_toggle);
    } else {
      this.state.compact = createNodesAndEdges(nextProps, this.state.full_toggle);
    }
  } 
  handleClick() {
    if (this.state.full_toggle) {
      this.setState(prevState => ({
        full_toggle: !prevState.full_toggle,
        compact: createNodesAndEdges(this.props,!this.state.full_toggle)
      }));
    } else {
      this.setState(prevState => ({
        full_toggle: !prevState.full_toggle,
        full: createNodesAndEdges(this.props,!this.state.full_toggle)
      }));
    }
  }
  render() {
    if (this.state.full_toggle) {
      if (this.state.full.nodes.length != 0 && 'count' in this.state.full.nodes[this.state.full.nodes.length-1]) {
        return (
          <div style={{position: "relative"}}>
            <ToggleButton onClick={this.handleClick}>Toggle view</ToggleButton>
            <CreateGraph nodes={this.state.full.nodes} edges={this.state.full.edges}/>
          </div>
        );
      } 
    } else {
      if (this.state.compact.nodes.length != 0 && 'count' in this.state.compact.nodes[this.state.compact.nodes.length-1]) {
        return (
          <div style={{position: "relative"}}>
            <ToggleButton onClick={this.handleClick}>Toggle view</ToggleButton>
            <CreateGraph nodes={this.state.compact.nodes} edges={this.state.compact.edges}/>
          </div>
        );
      }
    } 
    return null;
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    'dictionary': state.submission.dictionary,
    'counts_search': state.submission.counts_search,
    'links_search': state.submission.links_search,
  };
}

const mapDispatchToProps = (dispatch) => {
  return {
    onGetCounts: (type, project) => dispatch(getCounts(type, project)),
  };
}
const DataModelGraph = connect(mapStateToProps, mapDispatchToProps)(DataModelGraphComponent);
export default DataModelGraph;
