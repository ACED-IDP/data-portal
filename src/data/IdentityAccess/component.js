import React from 'react';
import Nav from '../nav.js'
import { json_to_string, get_submit_path } from '../utils'
import { updatePopup } from '../actions';
import {Popup, SavePopup} from '../Popup';
import { connect } from 'react-redux';
import { fetchStorageAccess, createUser, createKey, deleteKey,
  requestDeleteKey, clearDeleteSession, clearCreationSession } from './actions';
import { Box } from '../../theme';
import { RequestButton, DeleteButton } from './style';
import * as constants from "./constants";

const Entity = ({value, onUpdatePopup, onRequestDeleteKey}) => {
  let onDelete = () => {
    onRequestDeleteKey(value.access_key_id);
    onUpdatePopup({key_delete_popup: true});
  };
  return (
    <tr>
      <td width="40%">{value.access_key}</td>
      <td width="10%" style={{'textAlign':'left'}}>
        <DeleteButton onClick={onDelete}>
          {constants.DELETE_BTN}
        </DeleteButton>
      </td>
    </tr>
  )
};

const Entities = ({values, onUpdatePopup, onRequestDeleteKey}) => {
  return (
    <table width="100%">
      <tbody>
        <tr>
          <th>{constants.ACCESS_KEY_ID_COLUMN}</th>
          <th></th>
        </tr>
        {values.map( (item) => <Entity key={item.access_key} value={item}
                                       onUpdatePopup={onUpdatePopup}
                                       onRequestDeleteKey={onRequestDeleteKey}/> )}
      </tbody>
    </table>
  )
};

const IdentityComponent = ({cloud_access, popups, onCreateUser, onCreateKey, onClearCreationSession, onUpdatePopup, onDeleteKey,
                             onRequestDeleteKey, onClearDeleteSession}) => {
  return  (
    <Box>
      <Nav />
      {
        !cloud_access.access_key_pairs === undefined &&
        <div>
          {constants.NO_ACCESS_MSG}
        </div>
      }
      {
        cloud_access.access_key_pairs !== undefined &&
        <div>
          <h3>{constants.ACCESS_KEY_LBL}</h3>
          {
            popups.key_delete_popup === true &&
            <Popup message={constants.CONFIRM_DELETE_MSG}
                   error={json_to_string(cloud_access.delete_error)}
                   onConfirm={()=>onDeleteKey(cloud_access.request_delete_key)}
                   onCancel={()=>{ onClearDeleteSession(); onUpdatePopup({key_delete_popup: false})}}/>
          }
          {
            popups.save_key_popup === true &&
            <SavePopup message={constants.SECRET_KEY_MSG}
                       error={json_to_string(cloud_access.create_error)}
                       display={cloud_access.access_key_pair}
                       savingStr={cloud_access.str_access_key_pair}
                       onClose={()=>{onUpdatePopup({save_key_popup: false}); onClearCreationSession()}}
                       filename={'accessKeys.txt'}
            />
          }
          <RequestButton onClick={onCreateKey}>{constants.CREATE_ACCESS_KEY_BTN}</RequestButton>
          {
            cloud_access.access_key_pairs &&
            <Entities key='list_access_id' values={cloud_access.access_key_pairs}
                      onUpdatePopup={onUpdatePopup}
                      onRequestDeleteKey={onRequestDeleteKey}
            />
          }
        </div>
      }
    </Box>
  )
};


const mapStateToProps = (state) => {
  return {
    'cloud_access': state.cloud_access,
    'popups': state.popups
  }
};

const mapDispatchToProps = (dispatch) => {
  return {
    onCreateUser: (state) => dispatch(createUser()),
    onCreateKey: (state) => dispatch(createKey()),
    onUpdatePopup: (state) => dispatch(updatePopup(state)),
    onDeleteKey: (access_key) => dispatch(deleteKey(access_key)),
    onRequestDeleteKey: (access_key) => dispatch(fetchStorageAccess()).then(
        () => dispatch(requestDeleteKey(access_key))
    ),
    onClearDeleteSession: () => dispatch(clearDeleteSession()),
    onClearCreationSession: () => dispatch(clearCreationSession())
  };
};

let CloudAccess = connect(mapStateToProps, mapDispatchToProps)(IdentityComponent);
export default CloudAccess;
