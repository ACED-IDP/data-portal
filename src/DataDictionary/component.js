import React from 'react'
import Nav from '../Nav/component.js'
import {Box, cube, Table, TableData, TableRow, TableHead} from '../theme.js'
import { connect } from 'react-redux';
import styled from 'styled-components';
import * as constants from "./constants";
import { Link } from 'react-router';
import { BoxWithNavAndTimeout } from '../component';

const TableBullet = ({node, description})=>{
  return(
    <TableRow>
    <TableData>
      <Link to={'/dd/' + node }> {node} </Link>
    </TableData>
    <TableData right>
      {description}
    </TableData>
    </TableRow>
 )
}


const CategoryTable = ({dictionary, nodes, category}) =>{
  return(
  <Table>
    <TableHead>
      <tr>
        <td>
        {category}
        </td>
      </tr>
    </TableHead>

    <tbody>
      {nodes.map( (node) => <TableBullet node={node} key={node} description={dictionary[node]["description"]}/> )}
    </tbody>
  </Table>
 )
}

const DataDictionaryViewer = ({submission}) =>{
  let filterCategories = (dictionary, node_types) =>{
    let categories = {}
    for(let node in node_types){
      let category = dictionary[node_types[node]]["category"]
      if (category == undefined){
        continue;
      }
      if(categories.hasOwnProperty(category)){
        categories[category].push(node_types[node])
      }
      else{
        categories[category] = [node_types[node]]
      }
    }
    return categories
  }
  let categories = filterCategories(submission.dictionary, submission.node_types);
  return (
    <BoxWithNavAndTimeout>
      <h3> Data Dictionary Viewer </h3>
      <p>{constants.subHeader}</p>
      {Object.keys(categories).map((category) =>
        <CategoryTable dictionary={submission.dictionary} key={category} nodes={categories[category]} category = {category}/> )}
    </BoxWithNavAndTimeout>
  )
}

const mapStateToProps = (state)=> {return {
  'submission': state.submission,
}};

const mapDispatchToProps = (dispatch) => ({
});

const DataDictionary = connect(mapStateToProps, mapDispatchToProps)(DataDictionaryViewer);
export default DataDictionary
