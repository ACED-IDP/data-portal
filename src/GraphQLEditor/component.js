import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { BoxWithNavAndTimeout } from '../component';
import { fetchGraphQL } from '../actions';
import GraphiQL from 'graphiql';
import { buildClientSchema } from 'graphql/utilities';

export const Title = styled.h2`
  display: inline-block;
  vertical-align: middle;
  margin: 15px 0px;
  margin-right: 0.5em;
`;

let graphqlSchema = buildClientSchema(require('../data/schema.json').data);
let parameters = {};

const ProjectSubmissionComponent = (props) => {
  let editQuery = (newQuery) => {
    parameters.query = newQuery;
  };
  let editVariables = (newVariables) => {
    parameters.variables = newVariables;
  };

  return (
    <BoxWithNavAndTimeout>
      <div id="graphiql">
        <Title>Query graph</Title>
        <GraphiQL
          fetcher={fetchGraphQL}
          schema={graphqlSchema}
          query={parameters.query}
          variables={parameters.variables}
          onEditQuery={editQuery}
          onEditVariables={editVariables} />
      </div>
    </BoxWithNavAndTimeout>
  );
};

const mapStateToProps = (state, ownProps) => {
  return {
    'parameters': state.parameters
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
  };
};

const GraphQLQuery = connect(mapStateToProps, mapDispatchToProps)(ProjectSubmissionComponent);
export default GraphQLQuery;
