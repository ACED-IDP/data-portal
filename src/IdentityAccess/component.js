import React from 'react';
import { json_to_string, get_submit_path } from '../utils'
import { updatePopup } from '../actions';
import { Popup, SavePopup } from '../Popup/component';
import { connect } from 'react-redux';
import { fetchStorageAccess, createUser, createKey, deleteKey,
  requestDeleteKey, clearDeleteSession, clearCreationSession } from './actions';
import { Box } from '../theme';
import { RequestButton, DeleteButton, ProjectBullet } from './style';
import * as constants from "./constants";

const Entity = ({value, onUpdatePopup, onRequestDeleteKey}) => {
  let onDelete = () => {
    onRequestDeleteKey(value.access_key);
    onUpdatePopup({key_delete_popup: true});
  };
  return (
    <tr>
      <td width="40%">{value.access_key}</td>
      <td width="10%" style={{'textAlign':'left'}}>
        <DeleteButton onClick={onDelete}>
          {constants.DELETE_BTN}
        </DeleteButton>
      </td>
    </tr>
  )
};

const Entities = ({values, onUpdatePopup, onRequestDeleteKey}) => {
  return (
    <table width="100%">
      <tbody>
        <tr>
          {values.length > 0 &&
            <th>{constants.ACCESS_KEY_COLUMN}</th>
          }
        </tr>
        {values.map( (item) => <Entity key={item.access_key} value={item}
                                       onUpdatePopup={onUpdatePopup}
                                       onRequestDeleteKey={onRequestDeleteKey}/> )}
      </tbody>
    </table>
  )
};

const IdentityComponent = ({user, cloud_access, popups, onCreateUser, onCreateKey, onClearCreationSession, onUpdatePopup, onDeleteKey,
                             onRequestDeleteKey, onClearDeleteSession}) => {
  return  (
    <div>
      {
        !cloud_access.access_key_pairs === undefined &&
        <div>
          {constants.NO_ACCESS_MSG}
        </div>
      }
      {
        cloud_access.access_key_pairs !== undefined &&
        <div>
          {
            popups.key_delete_popup === true &&
            <Popup message={constants.CONFIRM_DELETE_MSG}
                   error={json_to_string(cloud_access.delete_error)}
                   onConfirm={()=>onDeleteKey(cloud_access.request_delete_key)}
                   onCancel={()=>{ onClearDeleteSession(); onUpdatePopup({key_delete_popup: false})}}/>
          }
          {
            popups.save_key_popup === true &&
            <SavePopup message={constants.SECRET_KEY_MSG}
                       error={json_to_string(cloud_access.create_error)}
                       display={cloud_access.access_key_pair}
                       savingStr={cloud_access.str_access_key_pair}
                       onClose={()=>{onUpdatePopup({save_key_popup: false}); onClearCreationSession()}}
                       filename={'accessKeys.txt'}
            />
          }
          <RequestButton onClick={onCreateKey}>{constants.CREATE_ACCESS_KEY_BTN}</RequestButton>
          {
            cloud_access.access_key_pairs &&
            <Entities key='list_access_id' values={cloud_access.access_key_pairs}
                      onUpdatePopup={onUpdatePopup}
                      onRequestDeleteKey={onRequestDeleteKey}
            />
          }
        </div>
      }
      <ul>
        <h4>{constants.LIST_PROJECT_MSG}</h4>
        {Object.keys(user.project_access).filter(
          (project) => user.project_access[project].indexOf('read-storage') != -1).map(
            (project, i) => <ProjectBullet key={i}>{project}</ProjectBullet>)}
      </ul>
    </div>
  )
};


const mapStateToProps = (state) => {
  return {
    'user': state.user,
    'cloud_access': state.cloud_access,
    'popups': state.popups
  }
};

const mapDispatchToProps = (dispatch) => {
  return {
    onCreateUser: (state) => dispatch(createUser()),
    onCreateKey: (state) => dispatch(createKey()),
    onUpdatePopup: (state) => dispatch(updatePopup(state)),
    onDeleteKey: (access_key) => dispatch(deleteKey(access_key)),
    onRequestDeleteKey: (access_key) => dispatch(fetchStorageAccess()).then(
        () => dispatch(requestDeleteKey(access_key))
    ),
    onClearDeleteSession: () => dispatch(clearDeleteSession()),
    onClearCreationSession: () => dispatch(clearCreationSession())
  };
};

let CloudAccess = connect(mapStateToProps, mapDispatchToProps)(IdentityComponent);
export default CloudAccess;
