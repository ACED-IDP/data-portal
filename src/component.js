import React from 'react';
import { connect } from 'react-redux';
import { Box } from './theme';
import Nav from './Nav/component';
import { TimeoutPopup } from './Popup/component';

export const BoxWithNavAndTimeout = ({children}) => {
  return (
    <Box>
      <Nav />
      <TimeoutPopup />
      { children }
    </Box>
  )
};
