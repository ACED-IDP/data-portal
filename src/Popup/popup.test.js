import React from 'react';
import { Popup } from './component';
import renderer from 'react-test-renderer';


test('Test Popup', () => {
  let onClose = () => {
    console.log('closed');
  }
  let onCancel = () => {
    console.log('canceled');
  }
  let onConfirm = () => {
    console.log('confirmed');
  }
  // react-hightlight doesn't work with jest snapshot, so code and error is not passed
  const component = renderer.create(
    <Popup message='test' onConfirm={onConfirm} onCancel={onCancel} onClose={onClose}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
