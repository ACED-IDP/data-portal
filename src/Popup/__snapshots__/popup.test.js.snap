// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Popup 1`] = `
<section
  className="sc-htpNat dDTzPz"
>
  <section
    className="sc-bxivhb jJjITh"
  >
    <div
      className="sc-EHOje gizWZa"
    >
      <div>
        test
      </div>
    </div>
    <a
      className="sc-bwzfXH eBwlJa"
      onClick={[Function]}
    >
      close
    </a>
    <a
      className="sc-bdVaJa hiblcO"
      onClick={[Function]}
    >
      confirm
    </a>
    <a
      className="sc-bwzfXH eBwlJa"
      onClick={[Function]}
    >
      cancel
    </a>
  </section>
</section>
`;
